### PR TITLE
feat(CommunityMintTokens): Sortable token holders list

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

- introduces `StatusSortableColumnHeader` component to `StatusQ` - reusable building block for creating sortable lists
- adds sortable token holders list component
- replaces the non-sortable holders list in `CommunityTokenView`
- basic integration of menu actions
- fixes some regressions related to remote destruct popups

Closes: https://github.com/status-im/status-desktop/issues/10602
Closes: https://github.com/status-im/status-desktop/issues/10621

### Affected areas
`CommunityTokenView`

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

[Screencast from 05.06.2023 17:15:59.webm](https://github.com/status-im/status-desktop/assets/20650004/1d4fc3cd-d142-45a6-92d8-7818fe1d5970)

[Screencast from 05.06.2023 17:23:32.webm](https://github.com/status-im/status-desktop/assets/20650004/616f2c56-f817-4afa-893b-5ce3c430c779)

Screenshot from 2023-06-05 17-20-09

### CODE REVIEW

File: ui/app/AppLayouts/Chat/panels/communities/NoHoldersPanel.qml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        StatusBaseText {
+            Layout.fillWidth: true
+
+            wrapMode: Text.Wrap
+            font.pixelSize: Style.current.primaryTextFontSize + 2
+            font.weight: Font.Bold
+
+            horizontalAlignment: Text.AlignHCenter
+            color: Theme.palette.directColor1
+            text: qsTr("No hodlers just yet")
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/panels/communities/NoHoldersPanel.qml:34]:
```suggestion
            text: qsTr("No holders just yet")
```

File: ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersList.qml [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+            RowLayout {
+                id: holdingHeader
+
+                Item {
+                    Layout.preferredWidth: 25
+                }
+
+                ColumnHeader {
+                    text: qsTr("Hodling")
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersList.qml:134]:
```suggestion
                    text: qsTr("Holding")
```

File: ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersList.qml [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+                spacing: Style.current.padding
+
+                StatusListItem {
+                    readonly property bool unknownHolder: model.name === ""
+                    readonly property string formattedTitle: unknownHolder
+                                                             ? "?" : model.name
+
+                    readonly property string walletAddress: model.walletAddress
+                    readonly property string walletAddressShortened:
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersList.qml:211]: I think we have a function to elide the wallet address

File: ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersPanel.qml [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+            Layout.topMargin: 12
+
+            visible: !root.empty
+
+            topPadding: 0
+            bottomPadding: 0
+            minimumHeight: 36 // by design
+            maximumHeight: minimumHeight
+
+            placeholderText: qsTr("Search hodlers")
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersPanel.qml:88]:
```suggestion
            placeholderText: qsTr("Search holders")
```

File: ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersPanel.qml [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+            Layout.topMargin: 12
+
+            wrapMode: Text.Wrap
+            font.pixelSize: Style.current.primaryTextFontSize
+            color: Theme.palette.baseColor1
+
+            visible: searcher.text.length > 0
+
+            text: (searcher.text.length > 0 && proxyModel.count > 0)
+                  ? qsTr("Search results") : qsTr("No hodlers found")
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersPanel.qml:102]:
```suggestion
                  ? qsTr("Search results") : qsTr("No holders found")
```

**Comment by caybro** [comment]
I hope I didn't miss any hodlers :smile:

**Comment by micieslak** [comment]
> I hope I didn't miss any hodlers smile

:tada:

https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5?node-id=29456:604507#460035532

Actually, we should check if we don't have any `hold*` :)

**Review by alexjba — APPROVED** [review]
Nice work!

**Review by noeliaSD — COMMENTED** [review]
It looks very nice!!! Great job! I didn't test the real flow, just `storybook` and checked minting tokens real flow **without** holders view.

Here the inspection review (will do the code review one in a second step):

- Does it make sense to have the `no holders box` also in the remotely destruct popup here?
Screenshot 2023-06-07 at 08 20 01

- The address text when we don't have information about the name of the holder is black in the design, grey in the implementation.

- The address when we know about the holder name is 14 px in design and in implementation seems to be the same size than name (15 px).

- It seems the number of messages (not sure if also the holdings) need formatting:
Design
Screenshot 2023-06-07 at 08 39 24

Implementation 
Screenshot 2023-06-07 at 08 39 41

_- Regarding context menu:_
1. Following the design, Airdrop and remotely destruct should also have inside the label, the name of the token. I.e: `Remotely destruct Anniversary` or `Airdrop Anniversary`.
Screenshot 2023-06-07 at 08 26 02

2. The `Remotely destruct` option should not appear when we are in `Airdrop view`:
Screenshot 2023-06-07 at 08 28 16

3. Should the element keep the selected look once the context menu is open? In the design it looks like this, but in the implementation it removes the selection look:
Design:
Screenshot 2023-06-07 at 08 34 32

Implementation:
Screenshot 2023-06-07 at 08 34 42

4. Question: left clicking into the holder does nothing. Should it open the context menu?

_General things:_

Please add tasks for the context menu options that are already not implemented and also for updating the remotely destruct popup with the new holders list.

File: ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersList.qml [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+                    asset.name: model.imageSource
+                    asset.isImage: true
+                    asset.isLetterIdenticon: unknownHolder
+                    asset.color: Theme.palette.userCustomizationColors[d.red2Color]
+                }
+
+                NumberCell {
+                    Layout.preferredWidth: root.headerItem.noOfMessagesHeaderWidth
+
+                    text: model.name ? model.noOfMessages : "-"
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersList.qml:234]: Maybe?
```suggestion
                    text: model.name ?  LocaleUtils.numberToLocaleString(model.noOfMessages) : "-"
```

File: ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersList.qml [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+                }
+
+                NumberCell {
+                    Layout.preferredWidth: root.headerItem.holdingHeaderWidth
+
+                    text: model.amount
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersList.qml:240]: Maybe?
```suggestion
                    text: LocaleUtils.numberToLocaleString(model.amount)
```

File: ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersList.qml [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+import QtQuick.Layouts 1.15
+import QtQuick.Controls 2.15
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+import utils 1.0
+
+StatusListView {
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersList.qml:11]: Maybe here you could also add info about the expected model roles.

File: ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersPanel.qml [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+            Layout.fillWidth: true
+            Layout.preferredHeight: contentHeight
+            Layout.topMargin: 20
+
+            model: proxyModel
+
+            onClicked: {
+                if (mouse.button !== Qt.RightButton)
+                    return
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersPanel.qml:127]: Wondering if here we should do some action too.

File: ui/app/AppLayouts/Chat/panels/communities/TokenHoldersProxyModel.qml [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+            roleName: "walletAddress"
+            sortOrder: root.sortOrder
+            priority: 1
+        },
+
+        RoleSorter {
+            enabled: root.sortBy === SortableTokenHoldersList.SortBy.NoOfMessages
+            roleName: "noOfMessages"
+            sortOrder: root.sortOrder
+        },
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/panels/communities/TokenHoldersProxyModel.qml:55]:
```suggestion
        },
        
```

**Review by noeliaSD — COMMENTED** [review]
Minor comments added!

File: ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersPanel.qml (hunk 9 above)
Comment (micieslak) [reply ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersPanel.qml:127]: According to figma, it's right-click menu: https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29673-619998&t=Rw4edL9LegNa9ez6-0

**Comment by noeliaSD** [comment]
> #11028

Thanks for addressing everything!! Just a small think that I've seen now..

Related to point 2.. `Remotely destruct` option should be only shown IF the minted token is deployed with the `remotely destruct` flag on!

**Review by noeliaSD — APPROVED** [review]

**Pull request merged**